test/listen.4.test.js

```diff
@@ -98,34 +105,46 @@ test('listen on localhost binds IPv4 and IPv6 - promise interface', async t => {
         url: getUrl(app, lookup)
       }, (err, response, body) => {
         if (err) { return reject(err) }
-        t.equal(response.statusCode, 200)
-        t.same(body.toString(), 'hello localhost')
+        t.assert.strictEqual(response.statusCode, 200)
+        t.assert.deepStrictEqual(body.toString(), 'hello localhost')
         resolve()
       })
     })
   }
 })
 
-test('listen on localhost binds to all interfaces (both IPv4 and IPv6 if present) - callback interface', t => {
+test('listen on localhost binds to all interfaces (both IPv4 and IPv6 if present) - callback interface', (t, done) => {
   dnsCb.lookup('localhost', { all: true }, (err, lookups) => {
     t.plan(2 + (3 * lookups.length))
-    t.error(err)
+    t.assert.ifError(err)
 
     const app = Fastify()
     app.get('/', async () => 'hello localhost')
     app.listen({ port: 0, host: 'localhost' }, (err) => {
-      t.error(err)
-      t.teardown(app.close.bind(app))
-
-      for (const lookup of lookups) {
-        sget({
-          method: 'GET',
-          url: getUrl(app, lookup)
-        }, (err, response, body) => {
-          t.error(err)
-          t.equal(response.statusCode, 200)
-          t.same(body.toString(), 'hello localhost')
-        })
+      t.assert.ifError(err)
+      t.after(() => app.close())
+
+      // Loop over each lookup and perform the assertions
+      if (lookups.length > 0) {
+        let completedRequests = 0
+
+        for (const lookup of lookups) {
+          sget({
+            method: 'GET',
+            url: getUrl(app, lookup)
+          }, (err, response, body) => {
+            t.assert.ifError(err)
+            t.assert.strictEqual(response.statusCode, 200)
+            t.assert.deepStrictEqual(body.toString(), 'hello localhost')
+            // Increase completed requests counter
+            completedRequests++
```

Comment:
Could you use the new `const { waitForCb } = require('./toolkit')` here?
It has been added yesterday I think 😄

Usages example:
https://github.com/fastify/fastify/pull/6021/files#diff-6a98dbdd7ca9fb8e0adfcc8787c950f65344cd9581437118754f4adb1a791f7dR75-R80

Reply:
I have updated the test, see commit here:  fae9eb8